schedule/schedule-appengine/src/main/client/jasify/admin/activity/admin-activity.js

```diff
@@ -183,6 +206,46 @@
                 vm.alert('danger', 'Failed: ' + r.statusText);
             }
         }
+
+        $scope.$watch(
+            // This function returns the value being watched.
+            function () {
+                return vm.activity.start;
+            },
+            // This is the change listener, called when the value returned from the above function changes
+            function (newValue, oldValue) {
+                if (newValue !== oldValue) {
+                    var offset = newValue.getTime() - oldValue.getTime();
+                    vm.activity.finish = new Date(vm.activity.finish.getTime() + offset);
+                }
+            }
+        );
+
```

Comment:
for future reference, the 1st parameter to $watch can be an expression or a function, so this could be declared as (I'm almost 100% sure)
```javascript
$scope.$watch( 'vm.activity.start', function(newValue, oldValue){});
```